app/services/projects/create_service.rb

```diff
@@ -3,26 +3,41 @@
 module Projects
   # Service used to Create Projects
   class CreateService < BaseService
-    attr_accessor :namespace_params
+    ProjectCreateError = Class.new(StandardError)
+    attr_accessor :namespace_params, :project, :namespace
 
     def initialize(user = nil, params = {})
       super(user, params)
       @namespace_params = @params.delete(:namespace_attributes)
+      @project = Project.new(params.merge(creator: current_user))
+      @namespace = Namespace.find_by(id: namespace_params[:parent_id] || namespace_params[:parent])
     end
 
-    def execute
-      @project = Project.new(params.merge(creator: current_user))
+    def execute # rubocop:disable Metrics/AbcSize
+      raise ProjectCreateError, I18n.t('services.projects.create.namespace_required') if namespace.nil?
 
-      @project.build_namespace(namespace_params.merge(owner: current_user))
+      unless allowed_to_modify_projects_in_namespace?(namespace)
+        raise ProjectCreateError,
+              I18n.t('services.projects.create.no_permission',
+                     namespace_type: namespace.class.model_name.human)
+      end
+      create_associations(project)
+      project
+    rescue Projects::CreateService::ProjectCreateError => e
+      project.errors.add(:base, e.message)
+      project
+    end
 
-      @project.save
+    def create_associations(project)
+      project.build_namespace(namespace_params.merge(owner: current_user))
+      project.save
 
-      Members::CreateService.new(current_user, @project.namespace, {
+      return unless user_has_namespace_maintainer_access?
```

Comment:
I think this might result in the creation of a Membership when the following is true:

User creates a Project in a Subgroup, where they have Maintainer role in the parent group, but Owner role in the Subgroup. So really you want to check that they don't have Owner role and they do have Maintainer role.

Reply:
Yeah looks like it would do that. Updated in 20e119965680ad3aa0f927bffc00468e33c0fff7